### PR TITLE
docs: add default `statusCode` fallback to prevent 'error' is possibly undefined warning in `error.vue`

### PR DESCRIPTION
**Pull request by edwinsamodra**
### 🔗 Linked issue

resolves #<issue-number>

### 📚 Description

When first using the default `error.vue` template, TypeScript raised a warning:

```
'__VLS_ctx.error' is possibly 'undefined'.
```

This happens because the `error` prop doesn’t define any fallback value, so TypeScript considers it optional.
To avoid that warning and prevent potential runtime issues when Nuxt renders an error without a full payload, a default object containing `{ statusCode: 500 }` is added.

This ensures the `error` prop is always defined, keeps the component stable, and maintains type safety without altering the UI or existing behavior.

Additional small improvements:

* Preserves the `NuxtError` type definition
* Provides a safer default for edge cases where `statusCode` is missing

Would you like help crafting tests for this PR? 😊

### CODE REVIEW

File: docs/2.directory-structure/1.app/3.error.md [hunk 1]
```diff
@@ -12,7 +12,10 @@ During the lifespan of your application, some errors may appear unexpectedly at
 import type { NuxtError } from '#app'
 
 const props = defineProps({
```
Comment (danielroe) [inline docs/2.directory-structure/1.app/3.error.md]:
```suggestion
const props = defineProps<{ error: NuxtError }>()
```

File: docs/2.directory-structure/1.app/3.error.md [hunk 2]
```diff
@@ -12,7 +12,10 @@ During the lifespan of your application, some errors may appear unexpectedly at
...
-  error: Object as () => NuxtError,
+  error: {
+    type: Object as () => NuxtError,
+    default: () => ({ statusCode: 500 })
+  }
 })
```
Comment (danielroe) [inline docs/2.directory-structure/1.app/3.error.md]:
```suggestion
```

File: docs/2.directory-structure/1.app/3.error.md [hunk 3]
```diff
@@ -11,8 +11,11 @@ During the lifespan of your application, some errors may appear unexpectedly at
...
 import type { NuxtError } from '#app'
 
-const props = defineProps({
-  error: Object as () => NuxtError,
+const props = defineProps<{ error: NuxtError }>()
+  error: {
+    type: Object as () => NuxtError,
+    default: () => ({ statusCode: 500 })
+  }
 })
```
Comment (danielroe) [inline docs/2.directory-structure/1.app/3.error.md]:
```suggestion
```

**Comment by danielroe** [comment]
> Would you like help crafting tests for this PR? 😊

🤔

please do review the output of LLMs before using it, please 🙏  (I've written a bit about it [here](https://roe.dev/blog/using-ai-in-open-source))

**Review by danielroe — APPROVED** [review]
thank you! ❤️
